hyppo/independence/friedman_rafsky.py

```diff
@@ -0,0 +1,206 @@
+            Input data matrices. ``x`` and ``y`` must have the same number of
+            rows. That is, the shapes must be ``(n, p)`` and ``(n, 1)`` where
+            `n` is the number of combined samples and `p` is the number of
+            dimensions. ``y`` is the array of labels corresponding to the two
+            samples, respectively.
+        Returns
+        -------
+        stat : float
+            The computed Friedman Rafsky statistic.
+        """
+        x = np.transpose(x)
+        labels = np.transpose(y)
+
+        MST_connections = MST(x, labels)
+        stat = self.num_runs(labels, MST_connections)
+
+        return stat
+
+    def test(
+        self,
+        x,
+        y,
+        reps=1000,
+        workers=1,
+        is_distsim=False,
+        perm_blocks=None,
+        random_state=None,
+    ):
+        r"""
+        Calculates the Friedman Rafsky test statistic and p-value.
+        Parameters
+        ----------
+        x,y : ndarray
+            Input data matrices. ``x`` and ``y`` must have the same number of
+            rows. That is, the shapes must be ``(n, p)`` and ``(n, 1)`` where
+            `n` is the number of combined samples and `p` is the number of
+            dimensions. ``y`` is the array of labels corresponding to the two
+            samples, respectively.
+        reps : int, default: 1000
+            The number of replications used to estimate the null distribution
+            when using the permutation test used to calculate the p-value.
+        workers : int, default: 1
+            The number of cores to parallelize the p-value computation over.
+            Supply ``-1`` to use all cores available to the Process.
+        perm_blocks : None or ndarray, default: None
+            Defines blocks of exchangeable samples during the permutation test.
+            If None, all samples can be permuted with one another. Requires `n`
+            rows. At each column, samples with matching column value are
+            recursively partitioned into blocks of samples. Within each final
+            block, samples are exchangeable. Blocks of samples from the same
+            partition are also exchangeable between one another. If a column
+            value is negative, that block is fixed and cannot be exchanged.
+        random_state : int, default: None
+            The random_state for permutation testing to be fixed for
+            reproducibility.
+        Returns
+        -------
+        stat : float
+            The computed Friedman Rafsky statistic.
+        pvalue : float
+            The computed Friedman Rafsky p-value.
+        null_dist : array
+            The null distribution of Friedman Rafsky test statistics.
+        """
+
+        stat, pvalue, null_dist = perm_test(
+            self.statistic, x, y, reps, workers, is_distsim, perm_blocks, random_state
+        )
+        x = np.transpose(x)
+        labels = np.transpose(y)
+        MST_connections = MST(x, labels)
```

Comment:
I'm confused what you are doing in 125-129. `perm_test` already returns the observed statistic